lib/livingstyleguide/document.rb

```diff
@@ -29,6 +31,65 @@ class LivingStyleGuide::Document < ::Tilt::Template
       end
     end
   end
+  def self.add_navigation(anchor)
+    @@navigation << anchor
+  end
+
+  def self.navigation
+    menu = '<nav class="lsg--navigation"><ul>'
+    index = 0
+    headline_closed         = true
+    sub_headline_closed     = true
+    sub_sub_headline_closed = true
+
+    while index < @@navigation.length
```

Comment:
The following code is hard to read. Try splitting it up into smaller methods with self-explaining names.